OpenRA.Mods.Common/Widgets/ObserverProductionIconsWidget.cs

```diff
@@ -77,43 +77,88 @@ public override void Draw()
-					continue;
+				var current = currentItems.OrderBy(q => q.RemainingTimeActual).First();
+				var queue = current.Queue;
 
-				var faction = queue.Trait.Actor.Owner.Faction.InternalName;
-				var actor = queue.Trait.AllItems().FirstOrDefault(a => a.Name == current.Item);
+				var actor = queue.AllItems().FirstOrDefault(a => a.Name == current.Item);
 				if (actor == null)
 					continue;
 
+				queueCol += 1;
+
 				var rsi = actor.TraitInfo<RenderSpritesInfo>();
-				var icon = new Animation(world, rsi.GetImage(actor, world.Map.Rules.Sequences, faction));
+				var icon = new Animation(world, rsi.GetImage(actor, world.Map.Rules.Sequences, queue.Actor.Owner.Faction.InternalName));
 				var bi = actor.TraitInfo<BuildableInfo>();
 				icon.Play(bi.Icon);
-				var location = new float2(RenderBounds.Location) + new float2(queue.i * (IconWidth + IconSpacing), 0);
-				WidgetUtils.DrawSHPCentered(icon.Image, location + 0.5f * iconSize, worldRenderer.Palette(bi.IconPalette), 0.5f);
+				var location = new float2(RenderBounds.Location) + new float2(queueCol * (IconWidth * 2 + IconSpacing), 2);
+				WidgetUtils.DrawSHPCentered(icon.Image, location + iconSize, worldRenderer.Palette(bi.IconPalette), 1f);
 
-				var pio = queue.Trait.Actor.Owner.PlayerActor.TraitsImplementing<IProductionIconOverlay>()
+				var pio = queue.Actor.Owner.PlayerActor.TraitsImplementing<IProductionIconOverlay>()
 					.FirstOrDefault(p => p.IsOverlayActive(actor));
 				if (pio != null)
-					WidgetUtils.DrawSHPCentered(pio.Sprite, location + 0.5f * iconSize + pio.Offset(0.5f * iconSize),
-						worldRenderer.Palette(pio.Palette), 0.5f);
+					WidgetUtils.DrawSHPCentered(pio.Sprite, location + iconSize + pio.Offset(iconSize),
+						worldRenderer.Palette(pio.Palette), 1f);
 
-				var clock = clocks[queue.Trait];
+				var clock = clocks[queue];
 				clock.PlayFetchIndex(ClockSequence,
 					() => current.TotalTime == 0 ? 0 : ((current.TotalTime - current.RemainingTime)
 					* (clock.CurrentSequence.Length - 1) / current.TotalTime));
 				clock.Tick();
-				WidgetUtils.DrawSHPCentered(clock.Image, location + 0.5f * iconSize, worldRenderer.Palette(ClockPalette), 0.5f);
+				WidgetUtils.DrawSHPCentered(clock.Image, location + iconSize, worldRenderer.Palette(ClockPalette), 1f);
 
 				var tiny = Game.Renderer.Fonts["Tiny"];
 				var text = GetOverlayForItem(current, timestep);
 				tiny.DrawTextWithContrast(text,
-					location + new float2(16, 16) - new float2(tiny.Measure(text).X / 2, 0),
+					location + iconSize - new float2(tiny.Measure(text).X / 2, 2),
 					Color.White, Color.Black, 1);
+
+				if (currentItems.Count() > 1)
+				{
+					var bold = Game.Renderer.Fonts["Bold"];
+					text = currentItems.Count().ToString();
+					bold.DrawTextWithContrast(text,
+						new float2(RenderBounds.Location) + new float2(queueCol * (IconWidth * 2 + IconSpacing) + 5, 5),
+						Color.White, Color.Black, 1);
+				}
+			}
+		}
+
+		int SortGroupByInfoType(ProductionItem queue)
```

Comment:
For D2K someone mentioned it was in different other than other mods. I could keep it like it was, however it will be now more difficult with that more general code for multique support. As there are many buildings in various orders and I implemented to display in this order:
building, defence, infantry, vehicle. With sort-queues-in-their-order-of-enabling it won't be sorted.